api/src/main/java/org/openmrs/api/OrderExceptions.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import static org.openmrs.Order.Action.DISCONTINUE;
+
+/**
+ * Represents specific error conditions that are part of OrderService.
+ * The specific error classes define only a default constructor if they are meant to be used with only a single error message.
+ * In other cases, there should be static factory methods that can be used to create the error class with the specific error message needed.
+ */
+public class OrderExceptions {
```

Comment:
will obviously defer to you on whats best.
But here is why I made the decision not to make them there own classes.
1. There is a lot of exceptions being created and it would pollute the number of classes in that package.
2. These exceptions are all similar to each other and it might make it easier on user to see a list of all different Order Exception classes.
3. Unless static import used, when throwing them, the name is very expressive. new OrderExceptions.CannotEditExistingOrderException()

Downsides:
1. Not consistent with rest of project
2. OrderExceptions is essentially a package name. Making a separate package called orderexceptions would help alleviate concern 1 and 2.